src/pages/index.js

```diff
@@ -127,20 +182,77 @@ function handleProfileEditClick() {
 }
 
 function handleEditProfileSubmit(inputValues) {
-  userInfo.setUserInfo(profileTitleInput.value, profileDescriptionInput.value);
-  profileEditPopup.close();
+  profileEditPopup.showLoading();
+  api
+    .editProfileInfo(inputValues.title, inputValues.description)
+    .then(() => {
+      userInfo.setUserInfo(inputValues.title, inputValues.description);
+      profileEditPopup.hideLoading();
+    })
+    .catch(console.error)
+    .finally(() => {
+      profileEditPopup.close();
+    });
 }
 
 function handleAddCardSubmit(inputValues) {
-  const newCardData = {
-    name: addCardTitleInput.value,
-    link: addCardImageLinkInput.value,
-  };
-  const newCard = createCard(newCardData);
-  section.addItem(newCard);
-  addCardPopup.close();
+  addCardPopup.showLoading();
+  api
+    .createNewCard(inputValues.title, inputValues.link)
+    .then((res) => {
+      const newCard = createCard(res);
+      cardListSection.addItem(newCard);
+    })
+    .then((res) => {
+      addCardPopup.hideLoading();
+    })
+    .catch(console.error)
+    .finally(() => {
+      addCardPopup.close();
+    });
 }
 
 function handleCardClick(cardData) {
   previewImagePopup.open(cardData);
 }
+
+function handleDeleteBinClick(cardID) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    api
+      .deleteCard(cardID)
+      .then((res) => {
+        this.handleCardDelete();
```

Comment:
**Could be improved**

To have access to any method of the class in `index.js` you can pass `this` to the call:

```js

    this._cardDeleteButton.addEventListener("click", () => {
      this._handleDeleteBinClick(this._cardID, this);
    });


```


And now in `index.js` you can call the methods like this:


```js


function handleDeleteBinClick(cardID, cardInstance) {  // here you get `cardInstance`
  deleteCardPopup.open();
  deleteCardPopup.setSubmitAction(() => {
    api
      .deleteCard(cardID)
      .then((res) => {
        cardInstance.handleCardDelete();  // here you use `cardInstance`
      })
      .catch(console.error)
      .finally(() => {
        deleteCardPopup.close();
      });
  });
}
```